govcd/nsxt_distributed_firewall_test.go

```diff
@@ -0,0 +1,292 @@
+//go:build network || nsxt || functional || openapi || ALL
+// +build network nsxt functional openapi ALL
+
+package govcd
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	. "gopkg.in/check.v1"
+)
+
+// Test_NsxtDistributedFirewall creates a list of distributed firewall rules with randomized
+// parameters in two modes:
+// * System user
+// * Org Admin user
+func (vcd *TestVCD) Test_NsxtDistributedFirewallRules(check *C) {
+	skipNoNsxtConfiguration(vcd, check)
+	skipOpenApiEndpointTest(vcd, check, types.OpenApiPathVersion1_0_0+types.OpenApiEndpointEdgeGateways)
+
+	adminOrg, err := vcd.client.GetAdminOrgByName(vcd.config.VCD.Org)
+	check.Assert(adminOrg, NotNil)
+	check.Assert(err, IsNil)
+
+	nsxtExternalNetwork, err := GetExternalNetworkV2ByName(vcd.client, vcd.config.VCD.Nsxt.ExternalNetwork)
+	check.Assert(nsxtExternalNetwork, NotNil)
+	check.Assert(err, IsNil)
+
+	vdc, vdcGroup := test_CreateVdcGroup(check, adminOrg, vcd)
+	check.Assert(vdc, NotNil)
+	check.Assert(vdcGroup, NotNil)
+
+	// Run firewall tests as System user
+	fmt.Println("# Running Distributed Firewall tests as 'System' user")
+	test_NsxtDistributedFirewallRules(vcd, check, vdcGroup.VdcGroup.Id, vcd.client, vdc)
+
+	// Prep Org admin user and run firewall tests
+	userName := strings.ToLower(check.TestName())
```

Comment:
we do this like this, because when running the whole suite as ORG user then this test will be ignored?

Reply:
No. go-vcloud-director does not have an option to run tests as Org user at all so I wanted to test this out here.
The `strings.ToLower` is here because VCD will create a user in lowercase by default and user creation will fail.

Reply:
Thank you for answer.